addons/sale/sale.py

```diff
@@ -117,6 +120,7 @@ def _compute_tax_id(self):
     pricelist_id = fields.Many2one('product.pricelist', string='Pricelist', required=True, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="Pricelist for current sales order.")
     currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
     project_id = fields.Many2one('account.analytic.account', 'Analytic Account', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="The analytic account related to a sales order.", copy=False)
+    related_project_id = fields.Many2one('account.analytic.account', inverse='_inverse_project_id', related='project_id', string='Analytic Account', help="The analytic account related to a sales order.", copy=False)
```

Comment:
we have this usability case:
Sale Order in 'sale' state:
        If Analytic Account not set
            Analytic Account should not be Read only
        If Analytic Account set:
            Analytic Account Should be Read only.

here In form view of sale order, we need to declare project_id field two times with read-only attrs base on above condition.

but our limitation is that we can declare field only one time in view. so we added related field with following logic in view side:
-> invisible project_id field in 'sale' state. This field readonly in 'done' and 'cancel' state.
-> display related_project_id field only in 'sale' state and apply readonly attrs like if project_id is set in 'sale' state, than this field will be read-only.